modules/utxo-lib/test/integration_local_rpc/parse.ts

```diff
@@ -27,26 +33,90 @@ function runTestParse(network: Network, txType: FixtureTxType, scriptType: Scrip
   }
 
   const fixtureName = `${txType}_${scriptType}.json`;
+  describe(fixtureName, function () {
+    let fixture: TransactionFixtureWithInputs;
+    let parsedTx: Transaction;
+
+    before(async function () {
+      fixture = await readFixture(network, fixtureName);
+      parsedTx = utxolib.Transaction.fromBuffer(Buffer.from(fixture.transaction.hex, 'hex'), network);
+    });
+
+    it(`round-trip`, function () {
+      assert.strictEqual(typeof fixture.transaction.hex, 'string');
+      assert.strictEqual(parsedTx.toBuffer().toString('hex'), fixture.transaction.hex);
+    });
+
+    it(`parseSignatureScript`, function () {
+      parsedTx.ins.forEach((input, i) => {
+        const result = parseSignatureScript(input);
+
+        if (txType === 'deposit') {
+          return;
+        }
 
-  it(`round-trip ${fixtureName}`, async function () {
-    const rpcTx: RpcTransaction = await readFixture(network, fixtureName);
-    assert.strictEqual(typeof rpcTx.hex, 'string');
-    const buf = Buffer.from(rpcTx.hex, 'hex');
-    const parsedTx = utxolib.Transaction.fromBuffer(buf, network);
-    assert(parsedTx.toBuffer().equals(buf));
+        assert.strict(result.publicKeys !== undefined);
+        assert.strictEqual(result.publicKeys.length, 3);
+
+        switch (scriptType) {
+          case 'p2sh':
+          case 'p2shP2wsh':
+            assert.strictEqual(result.inputClassification, 'scripthash');
+            break;
+          case 'p2wsh':
+            assert.strictEqual(result.inputClassification, 'witnessscripthash');
+            break;
+        }
+      });
+    });
+
+    if (txType === 'deposit') {
```

Comment:
Is this `if` statement supposed to be here or in the `it` block below?

Reply:
For deposits we don't even want to set up the `verifySignatures` test.

If I return inside `it` it might look like the test actually executed successfully.